src/components/cursor-controller.js

```diff
@@ -3,383 +3,182 @@ const TARGET_TYPE_INTERACTABLE = 2;
-    return true;
-  },
-
-  handleTouchMove: function(touch) {
-    this.mousePos.set(touch.clientX / window.innerWidth * 2 - 1, -(touch.clientY / window.innerHeight) * 2 + 1);
-  },
-
-  handleTouchEnd: function() {
-    this.data.cursor.emit("cursor-release", {});
   },
 
-  handleMouseDown: function() {
+  startInteraction: function() {
     if (this._isTargetOfType(TARGET_TYPE_INTERACTABLE_OR_UI)) {
       this.data.cursor.emit("cursor-grab", {});
       return true;
-    } else if (this.inVR || this.isMobile) {
-      this._startTeleport();
-      return;
     }
     return false;
   },
 
-  handleMouseMove: function(e) {
-    this.mousePos.set(e.clientX / window.innerWidth * 2 - 1, -(e.clientY / window.innerHeight) * 2 + 1);
+  moveCursor: function(x, y) {
+    this.mousePos.set(x, y);
   },
 
-  handleMouseUp: function() {
+  endInteraction: function() {
     this.data.cursor.emit("cursor-release", {});
-    this._endTeleport();
   },
 
-  handleMouseWheel: function(e) {
-    if (this._isGrabbing()) {
-      switch (e.deltaMode) {
-        case e.DOM_DELTA_PIXEL:
-          this.currentDistanceMod += e.deltaY / 500;
-          break;
-        case e.DOM_DELTA_LINE:
-          this.currentDistanceMod += e.deltaY / 10;
-          break;
-        case e.DOM_DELTA_PAGE:
-          this.currentDistanceMod += e.deltaY / 2;
-          break;
-      }
-    }
-  },
-
-  _handleEnterVR: function() {
-    this.inVR = true;
-    this._updateController();
-  },
-
-  _handleExitVR: function() {
-    this.inVR = false;
-    this._updateController();
-  },
-
-  _handlePrimaryDown: function(e) {
-    if (e.target === this.controller || e.target === this.data.playerRig) {
-      const isInteractable = this._isTargetOfType(TARGET_TYPE_INTERACTABLE) && !this.grabStarting;
-      if (isInteractable || this._isTargetOfType(TARGET_TYPE_UI)) {
-        this.grabStarting = true;
-        this.data.cursor.emit("cursor-grab", e.detail);
-      } else if (e.type !== this.data.grabEvent) {
-        this._startTeleport();
-      }
+  changeDistanceMod: function(delta) {
+    const { minDistance, maxDistance } = this.data;
+    const targetDistanceMod = this.currentDistanceMod + delta;
+    if (this.currentDistance - targetDistanceMod > maxDistance) {
+      return;
     }
-  },
-
-  _handlePrimaryUp: function(e) {
-    if (e.target === this.controller || e.target === this.data.playerRig) {
-      this.grabStarting = false;
-      if (this._isGrabbing() || this._isTargetOfType(TARGET_TYPE_UI)) {
-        this.data.cursor.emit("cursor-release", e.detail);
-      } else if (e.type !== this.data.releaseEvent) {
-        this._endTeleport();
-      }
+    if (this.currentDistance - targetDistanceMod < minDistance) {
+      return;
     }
-  },
-
-  _handleModelLoaded: function() {
-    this.physicalHand = this.data.playerRig.querySelector(this.data.physicalHandSelector);
+    this.currentDistanceMod = targetDistanceMod;
   },
 
   _handleCursorLoaded: function() {
```

Comment:
The reason is because `this.data.cursor.object3DMap.mesh` is undefined when the `cursor-controller` `init`s, and we want to run:
`    this.data.cursor.object3DMap.mesh.renderOrder = window.APP.RENDER_ORDER.CURSOR;`